src/Microsoft.AspNetCore.Razor.Runtime/Runtime/TagHelpers/TagHelperDescriptorFactory.cs

```diff
@@ -235,26 +238,71 @@ private static IEnumerable<string> GetAllowedChildren(Type type, ErrorSink error
             IEnumerable<string> allowedChildren,
             TagHelperDesignTimeDescriptor designTimeDescriptor)
         {
-            var requiredAttributes = GetCommaSeparatedValues(targetElementAttribute.Attributes);
+            var requiredAttributeDescriptors = GetRequiredAttributeDescriptors(targetElementAttribute);
 
             return BuildTagHelperDescriptor(
                 targetElementAttribute.Tag,
                 typeName,
                 assemblyName,
                 attributeDescriptors,
-                requiredAttributes,
+                requiredAttributeDescriptors,
                 allowedChildren,
                 targetElementAttribute.ParentTag,
                 targetElementAttribute.TagStructure,
                 designTimeDescriptor);
         }
 
+        private static IEnumerable<TagHelperRequiredAttributeDescriptor> GetRequiredAttributeDescriptors(
+            HtmlTargetElementAttribute targetElementAttribute)
+        {
+            var separatedAttributes = GetCommaSeparatedValues(targetElementAttribute.Attributes);
+            return separatedAttributes.Select(attribute =>
+            {
+                var requiredAttributeDescriptor = new TagHelperRequiredAttributeDescriptor
+                {
+                    IsCSSSelector = attribute[0] == '['
+                };
+
+                var potentialSelector = default(char);
+                if (requiredAttributeDescriptor.IsCSSSelector)
+                {
+                    var equalsIndex = attribute.IndexOf('=');
+                    requiredAttributeDescriptor.Name = GetCSSAttributeName(attribute, equalsIndex);
+
+                    var afterEquals = attribute[equalsIndex + 1];
+                    var quoteOffset = afterEquals == '\'' || afterEquals == '"' ? 1 : 0;
+                    var valueStart = equalsIndex + 1 + quoteOffset;
+                    var valueLength = attribute.Length - equalsIndex - quoteOffset * 2 - 2;
+                    requiredAttributeDescriptor.Value = attribute.Substring(valueStart, valueLength);
```

Comment:
Man, I could have swore that whitespace after the equals messed things up; after trying again looks like all is good. Will fix.